src/Forms/TextInput.tsx

```diff
@@ -0,0 +1,105 @@
+import React, {
+  ReactElement,
+  useContext,
+  ChangeEventHandler,
+  FunctionComponent,
+} from 'react';
+import styled, { ThemeContext } from 'styled-components';
+import { BaseTheme } from 'Theme';
+import ValidationErrorMessage from './ValidationErrorMessage';
+
+/** Represents the possible values for TextInput's label positioning */
+enum POSITION {
+  TOP = 'top',
+  LEFT = 'left',
+}
+
+export interface TextInputProps {
+  /** Function to call on change event */
+  onChange: ChangeEventHandler;
+  /** The id of the text input field */
+  id: string;
+  /** The name of the text input field */
+  name: string;
+  /** The placeholder value of the input field */
+  placeholder?: string;
+  /** Specifies the type of input */
+  type?: string;
+  /** The current value in the text input field */
+  value: string;
+  /** Specifies the size of the font for the text in the input field */
+  fontSize?: string;
+  /** The body of the error message, if applicable */
+  errorMessage?: string;
+  /** Specifies the label text */
+  label: string;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** If true, text input field will be disabled */
+  disabled?: boolean;
+}
+
+const StyledTextInput = styled.input<TextInputProps>`
+  border: ${({ theme }): string => (theme.border.light)};
+  margin: 2px;
+`;
+
+const StyledLabelText = styled.span<{isLabelVisible: boolean}>`
+  display: ${({ isLabelVisible }): string => (isLabelVisible ? 'inline' : 'none')};
+`;
+
+const StyledLabel = styled.label<{labelPosition: POSITION}>`
+  display: flex;
+  flex-direction: ${({ labelPosition }): string => (labelPosition === POSITION.TOP ? 'column' : 'row')};
+`;
+
+const TextInput: FunctionComponent<TextInputProps> = (props): ReactElement => {
+  const {
+    onChange,
+    id,
+    type,
+    name,
+    placeholder,
+    value,
+    errorMessage,
+    label,
+    labelPosition,
+    isLabelVisible,
+    disabled,
+  } = props;
+  const theme: BaseTheme = useContext(ThemeContext);
+  return (
+    <StyledLabel htmlFor={id} labelPosition={labelPosition}>
+      <StyledLabelText isLabelVisible={isLabelVisible}>{label}</StyledLabelText>
+      <StyledTextInput
+        onChange={onChange}
+        id={id}
+        name={name}
+        placeholder={placeholder}
+        type={type}
+        theme={theme}
+        value={value}
+        label={label}
+        disabled={disabled}
+      />
+      {errorMessage
```

Comment:
I think this error message makes our layout slightly more complicated than we'd like. The error should appear dynamically, so in a `POSITION.LEFT` layout it's going to pop-up on the right of the input, squeezing the size of the input and label. I think we might want to use some kind of two-dimensional layout for this, either a `grid` or nesting a second `flex` for the label and error. Offhand, I'm thinking something like this, though we probably want to double-check with Vittorio and talk though how we want to set this up.

```
POSITION.TOP
|---------------|-------------------------------|
|      LABEL    |                      ERROR    |
|-----------------------------------------------|
|                     INPUT                     |
|-----------------------------------------------|

POSITION.LEFT
|---------------|-------------------------------|
|      LABEL    |                               |
|---------------|             INPUT             |
|      ERROR    |                               |
|---------------|-------------------------------|
```

Reply:
Replaced use of Flex box with CSS Grid through dd1cc2d

Reply:
Here's the new format of the TextInput variations:
```
POSITION.TOP
|-----------------------------------------------|
|                   LABEL                        
|-----------------------------------------------|
|                   INPUT                     
|-----------------------------------------------|
|                   ERROR                     
|-----------------------------------------------|

POSITION.LEFT
|---------------|-------------------------------|
|      LABEL    |            INPUT              |
|-----------------------------------------------|
|               |             ERROR             |
|---------------|-------------------------------|
```